extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/mappers/EdcApiExceptionMapper.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.exception.mappers;
+
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+import org.eclipse.dataspaceconnector.api.exception.AuthorizationFailedException;
+import org.eclipse.dataspaceconnector.api.exception.NotAuthorizedException;
+import org.eclipse.dataspaceconnector.api.exception.ObjectExistsException;
+import org.eclipse.dataspaceconnector.api.exception.ObjectNotFoundException;
+import org.eclipse.dataspaceconnector.api.exception.ObjectNotModifiableException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Provider
```

Comment:
Would be better to register it to the `resourceConfig`explicitly?
It will make this process more readable (those kind of annotation hide details and that's not a great thing)